code/modules/clothing/glasses/engine_goggles.dm

```diff
@@ -79,17 +84,38 @@
 				flick_sonar(O)
 
 /obj/item/clothing/glasses/meson/engine/proc/flick_sonar(obj/pipe)
-	if(ismob(loc))
-		var/mob/M = loc
-		var/image/I = new(loc = get_turf(pipe))
-		var/mutable_appearance/MA = new(pipe)
+	var/mob/M = loc
+	var/image/I = new(loc = get_turf(pipe))
+	var/mutable_appearance/MA = new(pipe)
+	MA.alpha = 128
+	I.appearance = MA
+	flick_overlay(I, list(M.client), 8)
+
+/obj/item/clothing/glasses/meson/engine/proc/show_rads()
+	var/mob/living/carbon/human/user = loc
+	var/list/rad_places = list()
+	for(var/datum/component/radioactive/thing in SSradiation.processing)
```

Comment:
This needs some limits, or when singulo get loose or similar it's going to get horribly expensive.

Reply:
Addressed

Reply:
how?

Reply:
I added a get_dist in the rad_places loop which I figure will be cheaper than in this loop since by starting the loop it's already copied the necessary list to be iterated.